src/modules/src/controller_lee_payload.c

```diff
@@ -1775,12 +1795,21 @@ void controllerLeePayload(controllerLeePayload_t* self, control_t *control, setp
       for (uint8_t j = 0; j < state->num_neighbors; ++j) {
         if (self->attachement_points[i].id == state->neighbors[j].id) {
           // this attachement point belongs to a neighbor
+          if (state->num_neighbors == 1) {
+            float az2 = setpoint->cableAngles[i].az;
+            float el2 = setpoint->cableAngles[i].el;
+            self->desiredCableUnitVec2 = computeUnitVec(az2, el2);
+          }
           found = true;
           break;
         }
       }
       if (!found) {
         attPoint = self->attachement_points[i].point;
+        float az = setpoint->cableAngles[i].az;
+        float el = setpoint->cableAngles[i].el;
+        self->desiredCableUnitVec = computeUnitVec(az, el);
+
```

Comment:
This doesn't make sure that the id's of setpoint and state match. It is not guaranteed that the order of cableAngles and attachement_points is the same. You likely need a new loop, rather than re-using this one.